### PR TITLE
Update hand wire guide

### PR DESCRIPTION
**Pull request by Gouty**
First round of changes to the hand wire guide

Changed the physical hand wiring section to be less prescriptive and hopefully show that there are multiple ways of skinning a cat (with examples).

Tried not to remove the content of the previous version but re-wrote and put some sections in collapsible `<details>` tags.

Added images (originally into the project but as per discussion in #1686 put them in an imgur album and linked)

(As an aside I think their should be a separate QMK documentation repo)

Still want to make some changes to the but figured it was in a good enough state now to merge.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* #1686 (partially)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by rishka** [comment]
I think a lot of these changes look good, but as someone who just went through this, I feel like some reference to the split keyboard feature should be mentioned. There are a couple of gotchas when hand-wiring a split keyboard that may not be as much of a problem when using a pcb based kit.

The main one that I have come across is when using i2c for connecting the halves, it will cause issues if the resistors are not present(i.e. in the other half, not connected). It may make more sense to have that in the split keyboard docs, but if so, then  maybe just a link to there.

Also, it may be nice to link to additional features as a next step, like RGB, Oled, etc.

File: docs/hand_wire.md [hunk 1]
```diff
@@ -100,30 +99,115 @@ Things act as they should! Which will get us the following data:
...
+![Example ISO matrix](https://i.imgur.com/UlJ4ZDP.png)
+
+Also bear in mind that the number of rows plus the number of columns can not exceed the number of I/O pins on your controller.  So the fullsize matrix shown above would be possible on a Proton C or Teensy++, but not on a regular Teensy or Pro Micro
+
+#### Common Microcontroller Boards
+
+| Board         | Controller    | # I/O  | Pinout |
+| :------------ |:-------------:| ------:| ------ |
+| Pro Micro     | ATmega32u4    |     20 | [link](https://learn.sparkfun.com/tutorials/pro-micro--fio-v3-hookup-guide/hardware-overview-pro-micro)       |
+| Teensy 2.0    | ATmega32u4    |     25 | [link](https://www.pjrc.com/teensy/pinout.html)       |
```
Comment (rishka) [inline docs/hand_wire.md]: Maybe add the Elite-c here?

File: docs/hand_wire.md [hunk 2]
```diff
@@ -169,34 +253,52 @@ Before beginning to solder, it helps to have your wire pre-bent (if using single
...
+
+Simple firmware can be created easily using the [Keyboard Firmware Builder](https://kbfirmware.com/) website.  Recreate your layout using [Keyboard Layout Editor](http://www.keyboard-layout-editor.com), import it and recreate the matrix (if not already done as part of [planning the matrix](#planning-the-matrix).
 
-To start out, download [the firmware](https://github.com/qmk/qmk_firmware/) - we'll be using my (Jack's) fork of TMK called QMK/Quantum. We'll be doing a lot from the Terminal/command prompt, so get that open, along with a decent text editor like [Sublime Text](http://www.sublimetext.com/) (paid) or [Visual Studio Code](https://code.visualstudio.com) (free).
+Go through the rest of the tabs, assigning keys until you get to the last one where you can compile and download your firmware.  The .hex file can be flashed straight onto your keyboard, and the .zip of source files can be modified for advanced functionality and compiled locally using the method described in the collapsable section below.
+
+
+<details>
+
+<summary>Creating and compiling your firmware locally (old method)</summary>
```
Comment (rishka) [inline docs/hand_wire.md]: Not sure that "old" is the right word for this
```suggestion
<summary>Creating and compiling your firmware locally (command line method)</summary>
```

Comment (rishka) [reply docs/hand_wire.md]: Also, you could link to https://docs.qmk.fm/#/newbs_getting_started for additional information

File: docs/hand_wire.md [hunk 3]
```diff
@@ -169,34 +253,52 @@ Before beginning to solder, it helps to have your wire pre-bent (if using single
 
 If you're not using any insulation, you can try to keep the column wires elevated, and solder them near the tips of the keyswitch contacts - if the wires are sturdy enough, they won't short out to the row wiring an diodes.
 
-## Wiring Things to the Teensy
+</details>
+
+# Wiring up the controller
+
```
Comment (rishka) [inline docs/hand_wire.md]: A section for existing firmware might be a nice to have here, if someone is building an existing handwire

File: docs/hand_wire.md (hunk 1 above)
Comment (drashna) [reply docs/hand_wire.md]: Elite C uses the same pinout as the Pro Micro, with some extra pins. And there is no official page for it.

File: docs/hand_wire.md [hunk 4]
```diff
@@ -319,6 +421,19 @@ Once everything is installed, running `make` in the terminal should get you some
 
 Once you have your `<project_name>.hex` file, open up the Teensy loader application, and click the file icon. From here, navigate to your `QMK/keyboards/<project_name>/` folder, and select the `<project_name>.hex` file. Plug in your keyboard and press the button on the Teensy - you should see the LED on the device turn off once you do. The Teensy Loader app will change a little, and the buttons should be clickable - click the download button (down arrow), and then the reset button (right arrow), and your keyboard should be ready to go!
 
+</details>
+
+## Flashing the Firmware
+
+Install the [QMK toolbox](https://github.com/qmk/qmk_toolbox).
```
Comment (noroadsleft) [inline docs/hand_wire.md]:
```suggestion
Install [QMK Toolbox](https://github.com/qmk/qmk_toolbox).
```

File: docs/hand_wire.md [hunk 5]
```diff
@@ -100,30 +99,115 @@ Things act as they should! Which will get us the following data:
...
+
+Also bear in mind that the number of rows plus the number of columns can not exceed the number of I/O pins on your controller.  So the fullsize matrix shown above would be possible on a Proton C or Teensy++, but not on a regular Teensy or Pro Micro
+
+#### Common Microcontroller Boards
+
+| Board         | Controller    | # I/O  | Pinout |
+| :------------ |:-------------:| ------:| ------ |
+| Pro Micro     | ATmega32u4    |     20 | [link](https://learn.sparkfun.com/tutorials/pro-micro--fio-v3-hookup-guide/hardware-overview-pro-micro)       |
+| Teensy 2.0    | ATmega32u4    |     25 | [link](https://www.pjrc.com/teensy/pinout.html)       |
+| QMK Proton C      | STM32F303xC   |     36 | [link 1](https://i.imgur.com/RhtrAlc.png), [2](https://deskthority.net/wiki/QMK_Proton_C)       |
```
[the remark this answers is not in the document]
Comment (Gouty) [reply docs/hand_wire.md]: It would, if the official page had a pinout

File: docs/hand_wire.md (hunk 1 above)
Comment (Gouty) [reply docs/hand_wire.md]: Added a note about the Elite C and made a reference to split keyboards with a link to the relevant page

File: docs/hand_wire.md (hunk 2 above)
Comment (Gouty) [reply docs/hand_wire.md]: Done!

File: docs/hand_wire.md (hunk 3 above)
Comment (Gouty) [reply docs/hand_wire.md]: I added this to the start of "planning the matrix"

File: docs/hand_wire.md [hunk 6]
```diff
@@ -100,30 +99,115 @@ Things act as they should! Which will get us the following data:
...
+## Planning the matrix
+
+What you want to achieve is one leg from each switch being attached to the corresponding switches next to it (rows) and the other leg being attached to the switches above and below it (columns) and a diode to one of the legs, mosy commonly this will be the leg attached to the rows, and the diode will face away from it (Column to Row) i.e. with the wire furthest from the black line on the diode connected to the switch (as current will only travel in one direction through a diode)
+
+It is fairly simple to plan for an ortholinear keyboard (like a planck) 
+
+![Example planck matrix](https://i.imgur.com/FRShcLD.png)
+Image from [RoastPotatoes' "How to hand wire a Planck"](https://blog.roastpotatoes.co/guide/2015/11/04/how-to-handwire-a-planck/)
+
+But the larger and more complicated your keyboard, the more complex the matrix.  [Keyboard Firmware Builder](https://kbfirmware.com/) can help you plan your matrix layout (shown here with a basic fullsize ISO keyboard imported from [Keyboard Layout Editor](http://www.keyboard-layout-editor.com)
```
[the remark this answers is not in the document]
Comment (Gouty) [reply docs/hand_wire.md]: Reworded

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**